### PR TITLE
Avoid menus/views rendering issue with #login,password param

### PR DESCRIPTION
**Pull request by fjouatte**

### CODE REVIEW

File: auth_oauth2/static/src/js/auth_oauth2.js [hunk 1]
```diff
@@ -58,4 +58,34 @@ openerp.auth_oauth2 = function(instance) {
         },
     });
 
+    instance.web.OauthWebClient = instance.web.WebClient.extend({
```
Comment (petrus-v) [inline auth_oauth2/static/src/js/auth_oauth2.js]: I'm a bit surprised you are not using include here, where OauthWebClient is instantiated them?

File: auth_oauth2/static/src/js/auth_oauth2.js [hunk 2]
```diff
@@ -58,4 +58,34 @@ openerp.auth_oauth2 = function(instance) {
...
+    
+        bind_hashchange: function() {
+            var self = this;
+            $(window).bind('hashchange', this.on_hashchange);
+
+            var state = $.bbq.getState(true);
+            if (state.hasOwnProperty('login,password')) {
+                delete state['login,password'];
+            }
```
Comment (petrus-v) [inline auth_oauth2/static/src/js/auth_oauth2.js:70]: it's look like you only add those four lines, you may move it to the top and call super:

``` javascript
bind_hashchange: function() {
    var state = $.bbq.getState(true);
    if (state.hasOwnProperty('login,password')) {
        delete state['login,password'];
    }
    this._super();
}
```

what do you think?

File: auth_oauth2/static/src/js/auth_oauth2.js [hunk 3]
```diff
@@ -58,7 +58,7 @@ openerp.auth_oauth2 = function(instance) {
         },
     });
 
-    instance.web.OauthWebClient = instance.web.WebClient.extend({
+    instance.web.WebClient = instance.web.WebClient.extend({
```
Comment (petrus-v) [inline auth_oauth2/static/src/js/auth_oauth2.js:61]: it's not a good practice, prefer `include` instead `extend`:

```
instance.web.WebClient.include({});

```

File: auth_oauth2/static/src/js/auth_oauth2.js (hunk 2 above)
Comment (fjouatte) [reply auth_oauth2/static/src/js/auth_oauth2.js:70]: Nope, won't work : super will override the state value

File: auth_oauth2/static/src/js/auth_oauth2.js (hunk 3 above)
Comment (fjouatte) [reply auth_oauth2/static/src/js/auth_oauth2.js:61]: What about https://github.com/anybox/odoo-auth/blob/master/auth_oauth2/static/src/js/auth_oauth2.js#L10 ?

File: auth_oauth2/static/src/js/auth_oauth2.js (hunk 2 above)
Comment (petrus-v) [reply auth_oauth2/static/src/js/auth_oauth2.js:70]: Yes you're right!

File: auth_oauth2/static/src/js/auth_oauth2.js (hunk 3 above)
Comment (petrus-v) [reply auth_oauth2/static/src/js/auth_oauth2.js:61]: it looks the same mistake!

**Comment by petrus-v** [comment]
this fix #1

**Comment by petrus-v** [comment]
@fjouatte Thanks a lot!

**Pull request merged**